Formula/openjdk.rb

```diff
@@ -33,87 +34,89 @@ class Openjdk < Formula
-      sha256 "91310200f072045dc6cef2c8c23e7e6387b37c46e9de49623ce0fa461a24623d"
+      url "https://download.java.net/java/GA/jdk15.0.2/0d1cfde4252546c6931946de8db48ee2/7/GPL/openjdk-15.0.2_linux-x64_bin.tar.gz"
+      sha256 "91ac6fc353b6bf39d995572b700e37a20e119a87034eeb939a6f24356fbcd207"
     end
   end
 
   def install
-    # Path to dual-arch JavaNativeFoundation.framework from Xcode
-    framework_path = File.expand_path(
-      "../SharedFrameworks/ContentDeliveryServices.framework/Versions/Current/itms/java/Frameworks",
-      MacOS::Xcode.prefix,
-    )
-
-    boot_jdk_dir = Pathname.pwd/"boot-jdk"
-    resource("boot-jdk").stage boot_jdk_dir
-    boot_jdk = boot_jdk_dir/"Contents/Home"
-    java_options = ENV.delete("_JAVA_OPTIONS")
-
-    # Inspecting .hgtags to find a build number
-    # The file looks like this:
-    #
-    # 1613004c47e9dc867a2c2c43d716533b1aaedc5f jdk-15.0.2+0
-    # cc4fdb537bc14734064a9a8eadb091fd1c12b36e jdk-15.0.2+1
-    # d24e907486b3f90691980b0dde01efca5840abc6 jdk-15.0.2+2
-    # dbb11e11955ad1240ba775ab0007a14547e14ce6 jdk-15.0.2+3
-    # 4c4a2eb7b19ecb31620e6bb120e40f8a5fd1737a jdk-15.0.2+4
-    # e431a9461b1356c4b763443e5333b3f4a8695eaf jdk-15.0.2+5
-    # d5977ee56509ceaa3d3c8e1aebbca76651358da4 jdk-15.0.2+6
-    # 38912b2a5bcb396c75f8707e300773c874327451 jdk-15.0.2+7
-    #
-    # Since openjdk has move their development from mercurial to git and GitHub
-    # this approach may need some changes in the future
-    #
-    version_to_parse = if Hardware::CPU.arm?
-      "16"
-    else
-      version
+    boot_jdk = Pathname.pwd/"boot-jdk"
+    resource("boot-jdk").stage boot_jdk
+    on_macos do
+      boot_jdk /= "Contents/Home"
     end
-    build = File.read(".hgtags")
-                .scan(/ jdk-#{version_to_parse}\+(.+)$/)
-                .map(&:first)
-                .map(&:to_i)
-                .max
-    raise "cannot find build number in .hgtags" if build.nil?
+    java_options = ENV.delete("_JAVA_OPTIONS")
 
     args = %W[
-      --without-version-pre
-      --without-version-opt
-      --with-version-build=#{build}
-      --with-toolchain-path=/usr/bin
-      --with-boot-jdk=#{boot_jdk}
+      --disable-warnings-as-errors
       --with-boot-jdk-jvmargs=#{java_options}
+      --with-boot-jdk=#{boot_jdk}
       --with-debug-level=release
-      --with-native-debug-symbols=none
       --with-jvm-variants=server
-      --with-sysroot=#{MacOS.sdk_path}
-      --with-extra-ldflags=-headerpad_max_install_names
-      --enable-dtrace
+      --with-native-debug-symbols=none
+      --with-vendor-bug-url=#{tap.issues_url}
+      --with-vendor-name=#{tap.user}
+      --with-vendor-url=#{tap.issues_url}
+      --with-vendor-version-string=#{tap.user}
+      --with-vendor-vm-bug-url=#{tap.issues_url}
+      --with-version-build=#{revision}
+      --without-version-opt
+      --without-version-pre
     ]
 
-    if Hardware::CPU.arm?
+    on_macos do
```

Comment:
```suggestion
    framework_path = nil
    on_macos do
```
Maybe try this to increase scope of variable.

Haven't actually tried it myself though. I did do a quick copy-paste test previously and was able to build ARM if `framework_path` is defined inside other `on_macos` section.

Reply:
Alright let's try that.